testkit/gnmiconfig/worker.go

```diff
@@ -0,0 +1,353 @@
+			return expectedVal, true
+		}
+	}
+	return nil, false
+}
+
+func verifyConfiguration(client pb.GNMIClient, expectedVals map[*pb.Path]*pb.TypedValue, timeout time.Duration) error {
+	for gnmiPath, expectedVal := range expectedVals {
+		// Fetch the updated config from device.
+		getRequest := &pb.GetRequest{
+			Path:     []*pb.Path{gnmiPath},
+			Encoding: pb.Encoding_JSON_IETF,
+			UseModels: []*pb.ModelData{{
+				Name:         "office-ap",
+				Organization: "Google, Inc.",
+				Version:      "0.1.0",
+			}},
+		}
+		ctx, _ := context.WithTimeout(context.Background(), timeout)
+		getResponse, err := client.Get(ctx, getRequest)
+
+		// Check deleted config. Verify they are actually removed.
+		if expectedVal == nil && err == nil {
+			return fmt.Errorf("still able to fetch the deleted config on %v", gnmiPath)
+		}
+
+		if err != nil {
+			return fmt.Errorf("gNMI get request failed for path: %v", err)
+		}
+
+		// Check updated config is correct.
+		if err := verifyGetResponse(getResponse, gnmiPath, expectedVal); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func verifyGetResponse(getResponse *pb.GetResponse, gnmiPath *pb.Path, expectedVal *pb.TypedValue) error {
+	if len(getResponse.Notification) != 1 {
+		return fmt.Errorf("incorrect notification number in GetResponse, actual = %d, expected = 1", len(getResponse.Notification))
+	}
+
+	notification := getResponse.Notification[0]
+	if len(notification.Delete) != 0 {
+		return fmt.Errorf("incorrect Delete number in GetResponse, actual = %d, expected = 0", len(notification.Delete))
+	}
+	if len(notification.Update) != 1 {
+		return fmt.Errorf("incorrect Update number in GetResponse, actual = %d, expected = 1", len(notification.Update))
+	}
+
+	update := notification.Update[0]
+	updatedPath := gnmiFullPath(notification.Prefix, update.Path)
+	if !reflect.DeepEqual(updatedPath, gnmiPath) {
+		return fmt.Errorf("incorrect gnmi path in GetResponse, actual = %v, expected = %v", updatedPath, gnmiPath)
+	}
+
+	return valEqual(gnmiPath, update.Val, expectedVal)
+}
+
+func valEqual(gnmiPath *pb.Path, actual *pb.TypedValue, expected *pb.TypedValue) error {
+	actualValue := actual.Value
+	expectedValue := expected.Value
+
+	// JSON value.
+	actualJsonValue, okA := actualValue.(*pb.TypedValue_JsonIetfVal)
+	expectedJsonValue, okE := expectedValue.(*pb.TypedValue_JsonIetfVal)
+	if okA && okE {
+		var actualJson, expectedJson interface{}
+		if err := json.Unmarshal(actualJsonValue.JsonIetfVal, &actualJson); err != nil {
+			return fmt.Errorf("invalid value %v: %v", string(actualJsonValue.JsonIetfVal), err)
+		}
+		if err := json.Unmarshal(expectedJsonValue.JsonIetfVal, &expectedJson); err != nil {
+			return fmt.Errorf("invalid value %v: %v", string(expectedJsonValue.JsonIetfVal), err)
+		}
+		if !reflect.DeepEqual(actualJson, expectedJson) {
+			log.Errorf("actual=\n%v", string(actualJsonValue.JsonIetfVal))
+			log.Errorf("expected=\n%v", string(expectedJsonValue.JsonIetfVal))
```

Comment:
Why do we need the two lines above? (the information is included in the err msg below).

Reply:
That was for debugging, forgot to remove them.
Removed.